aten/src/ATen/core/ivalue.h

```diff
@@ -620,6 +620,7 @@ struct CAFFE2_API IValue final {
   };
 
   using HashAliasedIValues = std::unordered_set<IValue, HashAliasedIValue, CompAliasedIValues>;
+  using HashAliasedIValueMap = std::unordered_map<IValue, IValue, HashAliasedIValue, CompAliasedIValues>;
```

Comment:
Why are we defining `HashAliasedIValueMap` to be the same exact alias as . `HashAliasedIValues`? Also, `CompAliasedIValues` compares based on aliasing, not identity, so two tensors which are not equal but alias will be stored as one value, which I don't think is what you want for this PR. I think you want a equality function that just calls into `IValue.is()`

Reply:
I feel we do want aliasing here? cc @bzinodev

Reply:
this is different from HashAliasedIValues because it is a Map and HashAliasedIValues is a set

Reply:
For example, 
self.a = torch.tensor([...])
self.b = self.a.view(2,3)

we will want self.a and self.b share the same Tensor.

Reply:
Something like:
```
self.a = torch.tensor([...])
self.b = self.a.transpose()
```
would give the wrong result